standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/ObjectStore.java

```diff
@@ -13272,15 +13280,16 @@ public int deleteScheduledExecutions(int maxRetainSecs) {
       return 0;
     }
     boolean committed = false;
-    try {
+    try (Query q = pm.newQuery(MScheduledExecution.class)) {
       openTransaction();
       int maxCreateTime = (int) (System.currentTimeMillis() / 1000) - maxRetainSecs;
-      Query q = pm.newQuery(MScheduledExecution.class);
       q.setFilter("startTime <= maxCreateTime");
       q.declareParameters("int maxCreateTime");
       long deleted = q.deletePersistentAll(maxCreateTime);
       committed = commitTransaction();
       return (int) deleted;
+    } catch (Exception e) {
```

Comment:
is this catch needed?

Reply:
```try (Query q = pm.newQuery(MScheduledExecution.class)) {``` throws a checked exception that can't be ignored.